### PR TITLE
Error Boundary

### LINKED ISSUE #2: Implement Error Boundary
_Error boundaries are React components that catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI instead of the component tree that crashed._ ([1](https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries))

TODO:

- Implement Error Boundary class
- Wrap the Component tree
- Add tests for the Error Boundary

### PR DESCRIPTION
**Pull request by clauxx**
closes #2

### CODE REVIEW

File: src/ErrorBoundary.tsx [hunk 1]
```diff
@@ -11,12 +12,28 @@ export class ErrorBoundary extends React.Component {
     return {hasError: true};
   }
 
+  private onPressClose = () => {
+    this.setState({hasError: false});
+  };
+
   render() {
     if (this.state.hasError) {
+      // TODO An alternative solution would be to navigate to an Error screen, gather
```
Comment (SBub) [inline src/ErrorBoundary.tsx:21]: You could pass a param and based on it display a set of buttons. But beware error boundary is displayed outside of the navigation. So it would not be that straightforward. 

I would display the `Error` component manually here.
But for the rest of the use cases I would setup of a sub store that holds error state. And a listener that navigates to the Error screen when an error is registered in the store.

File: src/components/Error.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    <View style={styles.imageContainer}>
+      <View style={styles.imagePadding} />
+      <Image
+        resizeMode={'contain'}
+        style={styles.image}
+        source={require('~/assets/images/error-planets.png')}
+      />
+    </View>
+    <View style={styles.textContainer}>
+      <Text style={styles.title}>{title}</Text>
```
Comment (SBub) [inline src/components/Error.tsx:26]: This can totally be separated into a component as this combination of Header and Paragraph appear across screens

Screen Shot 2020-04-27 at 11 20 43 AM
Screen Shot 2020-04-27 at 11 21 22 AM
Screen Shot 2020-04-27 at 11 18 44 AM

File: src/components/Btn.tsx [hunk 3]
```diff
@@ -57,18 +89,23 @@ const Btn: React.FC<PropsI> = (props) => {
...
 
 const styles = StyleSheet.create({
   container: {
-    paddingVertical: 16,
     width: '100%',
     borderRadius: 8,
     marginVertical: 5,
   },
+  largeBtn: {
+    paddingVertical: 16,
```
[the remark this answers is not in the document]
Comment (clauxx) [reply src/components/Btn.tsx:103]: Not following. Across the app there are two types of button sizes, where the padding is `16` and `12`.

File: src/ErrorBoundary.tsx [hunk 4]
```diff
@@ -11,12 +12,28 @@ export class ErrorBoundary extends React.Component {
...
+      // the details {@title, @description} from the Error object and send them as
+      // navigation props. Not sure how to handle rendering the Buttons in this case.
       return (
-        <ScreenContainer>
-          <Text>Oh no, something happened</Text>
-        </ScreenContainer>
+        <ErrorComponent
```
[the remark this answers is not in the document]
Comment (clauxx) [reply src/ErrorBoundary.tsx:25]: Not sure we should use `Error` as a name here either, since it could conflict with the Error `object` or just cause confusion. Could not get any further than `ErrorComponent` though. Do you have a suggestion on this one?

File: src/utils/fonts.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import {Colors} from './colors';
+
+export enum Fonts {
+  Regular = 'TTCommons-Regular',
+  Medium = 'TTCommons-Medium',
+  Light = 'TTCommons-Light',
+}
+
+export const TextStyle = {
+  middleTitleRegular: {
```
[the remark this answers is not in the document]
Comment (clauxx) [reply src/utils/fonts.ts:10]: Replaced using `Header` and `Paragraph`.

[the remark this answers is not in the document]
Comment (clauxx) [reply src/utils/fonts.ts:10]: I'm not sure whether you meant to leave the comment in the `TextStyle` section.

File: src/components/Header.tsx [hunk 6]
```diff
@@ -14,6 +14,7 @@ interface PropsI {
 const Header: React.FC<PropsI> = ({size = HeaderSizes.medium, children}) => {
   return (
     <Text
+      testID={'header'}
```
Comment (SBub) [inline src/components/Header.tsx:17]: no need to use brackets here

Comment (SBub) [reply src/components/Header.tsx:17]: What about tests itself? Do we use this prop anywhere?

File: src/components/Paragraph.tsx [hunk 7]
```diff
@@ -21,7 +21,11 @@ const Paragraph: React.FC<PropsI> = ({
 }) => {
   const sizeStyle =
     size === ParagraphSize.medium ? styles.medium : styles.large;
-  return <Text style={[styles.paragraph, {color}, sizeStyle]}>{children}</Text>;
+  return (
+    <Text testID={'paragraph'} style={[styles.paragraph, {color}, sizeStyle]}>
```
Comment (SBub) [inline src/components/Paragraph.tsx:25]: the same as above

File: src/utils/styles.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    },
+  }),
+};
+
+export const secondaryTextStyle = {
+  opacity: 0.8,
+  fontFamily: Fonts.Regular,
+};
+
+export const debug = {
```
[the remark this answers is not in the document]
Comment (clauxx) [reply src/utils/styles.ts:29]: I use it frequently to see the dimensions of the view while developing. Not used in production though.
